analysis/src/Xform.ml

```diff
@@ -0,0 +1,945 @@
+                     {
+                       textDocument =
+                         {uri = newFilePath |> Uri.toString; version = None};
+                       edits =
+                         [
+                           {
+                             newText = textForExtractedFile;
+                             range =
+                               {
+                                 start = {line = 0; character = 0};
+                                 end_ = {line = 0; character = 0};
+                               };
+                           };
+                         ];
+                     };
+                   TextDocumentEdit
+                     {
+                       textDocument = {uri = path; version = None};
+                       edits = [{newText = newTextInCurrentFile; range}];
+                     };
+                 ]);
+        ()
+      | _ -> ());
+      Ast_iterator.default_iterator.structure_item iterator structure_item
+    in
+
+    {Ast_iterator.default_iterator with structure_item}
+
+  let xform ~pos ~codeActions ~path ~printStandaloneStructure structure =
+    let changed = ref None in
+    let iterator = mkIterator ~pos ~path ~changed ~printStandaloneStructure in
+    iterator.structure iterator structure;
+    match !changed with
+    | None -> ()
+    | Some codeAction -> codeActions := codeAction :: !codeActions
+end
+
+module AddBracesToFn = struct
+  (* Add braces to fn without braces *)
+
+  let mkIterator ~pos ~changed =
+    (* While iterating the AST, keep info on which structure item we are in.
+       Printing from the structure item, rather than the body of the function,
+       gives better local pretty printing *)
+    let currentStructureItem = ref None in
+
+    let structure_item (iterator : Ast_iterator.iterator)
+        (item : Parsetree.structure_item) =
+      let saved = !currentStructureItem in
+      currentStructureItem := Some item;
+      Ast_iterator.default_iterator.structure_item iterator item;
+      currentStructureItem := saved
+    in
+    let expr (iterator : Ast_iterator.iterator) (e : Parsetree.expression) =
+      let bracesAttribute =
+        let loc =
+          {
+            Location.none with
+            loc_start = Lexing.dummy_pos;
+            loc_end =
+              {
+                Lexing.dummy_pos with
+                pos_lnum = Lexing.dummy_pos.pos_lnum + 1 (* force line break *);
+              };
+          }
+        in
+        (Location.mkloc "res.braces" loc, Parsetree.PStr [])
+      in
+      let isFunction = function
+        | {Parsetree.pexp_desc = Pexp_fun _} -> true
+        | _ -> false
+      in
+      (match e.pexp_desc with
+      | Pexp_fun (_, _, _, bodyExpr)
+        when Loc.hasPos ~pos bodyExpr.pexp_loc
+             && isBracedExpr bodyExpr = false
+             && isFunction bodyExpr = false ->
+        (* bodyExpr.pexp_attributes <- bracesAttribute :: bodyExpr.pexp_attributes; *)
```

Comment:
@cristianoc This is problematic and I commented it out for now to get things to build.

It seems `pexp_attributes` was made mutable in the rescript-vscode repo. I guess that's a hack that we don't want to take over into the compiler repo?

Reply:
We can have it in the compiler repo with a comment not to use it.

Reply:
Ok, that feels a bit hacky, but will certainly do the job and can maybe be cleaned up later?